### PR TITLE
Feature/multiple scenes

### PR DESCRIPTION
**Pull request by Dartemiss**
Whats new?

- multiple scenes now are posible
- file/ save scene save current scene
- file/ save scene as save current scene in a new file
- file / load scene opens a custom imgui file explorer to choose which scene to load

### CODE REVIEW

**Review by Pacasasgar — APPROVED** [review]
That new imgui window for loading scenes is very cool, nice work!

Little 'TOC' comments:

-We are changing a lot the `<ShowAllFiles>true</ShowAllFiles> - <ShowAllFiles>false</ShowAllFiles> ` and commiting it.  Maybe we should not?

-PanelPopups.h we could separate public variables from functions by just spacing one line:
_void CreateScript();
void SaveScene();
void LoadScene();
bool create_script_shown = false;
bool save_scene_shown = false;
bool load_scene_shown = false;_

I know those are pretty silly comments but I wanted to mention it just in case :D

**Comment by mariofv** [comment]
> That new imgui window for loading scenes is very cool, nice work!
> 
> Little 'TOC' comments:
> 
>....
> 
> -PanelPopups.h we could separate public variables from functions by just spacing one line:
> _void CreateScript(); void SaveScene(); void LoadScene(); bool create_script_shown = false; bool save_scene_shown = false; bool load_scene_shown = false;_
> 
> I know those are pretty silly comments but I wanted to mention it just in case :D

They are not silly, actually you are right in the declaration part. They should be separate with another visiblity declaration (private/public).

We must reach code cleanliness perfection, as it is written in the [holy code styling standards](https://github.com/OnionGalaxy/LittleOrionEngine/blob/master/CONTRIBUTING.md).

**Comment by Dartemiss** [comment]
Yep! Thanks for the feedback :D This evening i will change what you propose.

File: Engine/UI/Panel/PanelPopups.cpp [hunk 1]
```diff
@@ -65,4 +72,81 @@ void PanelPopups::CreateScript()
...
+	{
+		if (!(mask & (1 << i)))
+			continue;
+		char rootName[4] = { static_cast<char>('A' + i), ':', '\\', '\0' };
+		UINT type = GetDriveTypeA(rootName);
+		if (type == DRIVE_REMOVABLE || type == DRIVE_FIXED)
+			ret |= (1 << i);
+	}
+	return ret;
+}
```
Comment (raulgonzalezupc) [inline Engine/UI/Panel/PanelPopups.cpp]: very needed for debug proposals and to do different scenes, very cool :)

File: Engine/UI/Panel/PanelPopups.cpp [hunk 2]
```diff
@@ -65,4 +72,81 @@ void PanelPopups::CreateScript()
...
+		static char str1[128] = "";
+		ImGui::InputTextWithHint("", "enter scene name here (XXXX.scene format)", str1, IM_ARRAYSIZE(str1));
+
+		if (ImGui::Button("Save", ImVec2(120, 0)))
+		{
+			std::string name = str1;
+			std::size_t found = name.find_last_of(".");
+			if (found == std::string::npos || found == 0)
+			{
+				APP_LOG_ERROR("Invalid name for scene, it should be of format XXXX.scene");
```
Comment (raulgonzalezupc) [inline Engine/UI/Panel/PanelPopups.cpp]: maybe just putting "Invalid name for scene, introduce ".scene" extension", just a suggestion

**Review by raulgonzalezupc — APPROVED** [review]

**Pull request merged**